backend/src/services/solanaService.js

```
@@ -1,220 +1,387 @@
+        return null;
+    }
+
+    let tokenAmount = 0;
+    let sourceTokenAccount = null;
+    let destinationTokenAccount = null;
+
+    const transferInstruction = tx.transaction.message.instructions.find(
+        inst => inst.parsed?.type === 'transfer' && inst.programId === TOKEN_PROGRAM_ID.toBase58()
+    );
+
+    if (transferInstruction && transferInstruction.parsed && transferInstruction.parsed.info) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)